docker/buildkite/Dockerfile

```diff
@@ -37,3 +37,11 @@ RUN git config --global --add safe.directory /cadence
 # https://github.com/docker-library/golang/blob/c1baf037d71331eb0b8d4c70cff4c29cf124c5e0/1.4/Dockerfile
 RUN mkdir -p /cadence
 WORKDIR /cadence
+
+# Copy go mod dependencies and try to share the module download cache
+COPY go.* /cadence
+COPY internal/tools/go.* /cadence/internal/tools/
+COPY cmd/server/go.* /cadence/cmd/server/
+COPY common/archiver/gcloud/go.* /cadence/common/archiver/gcloud/
```

Comment:
Does a buildkite agent share the cache? At the same time, we have to be sure that the cache is unique per checksum of the copied folder.

Reply:
Unfortunately, no. But at least it is done during the Docker build phase. It does not change speed much, but provides cleaner logs in the test run.
Long term, we are migrating to Docker actions, where we can leverage the cache action to save dependencies.